tests/api/notify_test.py

```diff
@@ -0,0 +1,139 @@
+
+
+class TestZino1NotificationProtocolBuildNotifications:
+    def test_should_make_notifications_for_regular_changed_attrs(self, fake_event):
+        protocol = Zino1NotificationProtocol()
+        event_copy = fake_event.model_copy(deep=True)
+        event_copy.updated = now() + timedelta(seconds=5)
+
+        notifications = list(protocol.build_notifications(new_event=event_copy, old_event=fake_event))
+        assert notifications == [Notification(event_id=42, change_type="attr", value="updated")]
+
+    def test_should_make_notifications_for_log_changes(self, fake_event):
+        protocol = Zino1NotificationProtocol()
+        event_copy = fake_event.model_copy(deep=True)
+        event_copy.add_log("foo")
+
+        notifications = list(protocol.build_notifications(new_event=event_copy, old_event=fake_event))
+        assert Notification(event_id=42, change_type="log", value=1) in notifications
+
+    def test_should_make_notifications_for_history_changes(self, fake_event):
+        protocol = Zino1NotificationProtocol()
+        event_copy = fake_event.model_copy(deep=True)
+        event_copy.add_history("foo")
+
+        notifications = list(protocol.build_notifications(new_event=event_copy, old_event=fake_event))
+        assert Notification(event_id=42, change_type="history", value=1) in notifications
+
+    def test_should_make_notifications_for_state_changes(self, fake_event):
+        protocol = Zino1NotificationProtocol()
+        event_copy = fake_event.model_copy(deep=True)
+        event_copy.state = EventState.IGNORED
+
+        old, new = fake_event.state.value, event_copy.state.value
+        expected = f"{old} {new}"
+
+        notifications = list(protocol.build_notifications(new_event=event_copy, old_event=fake_event))
+        assert Notification(event_id=42, change_type="state", value=expected) in notifications
+
+
+class TestZino1NotificationProtocolBuildAndSendNotifications:
+    def test_should_send_notifications_only_to_tied_channels(self, event_loop, fake_event, changed_fake_event):
+        server = ZinoServer(loop=event_loop, state=ZinoState())
+        channel1 = Mock()
+        channel2 = Mock()
+        mock_api = Mock()
+
+        channel1.tied_to = mock_api
+        channel2.tied_to = None
+
+        server.notification_channels["a"] = channel1
+        server.notification_channels["b"] = channel2
+
+        Zino1NotificationProtocol.build_and_send_notifications(
+            server, new_event=changed_fake_event, old_event=fake_event
+        )
+        assert channel1.notify.called
+        assert not channel2.notify.called
+
+
+@pytest.fixture
+def fake_event() -> ReachabilityEvent:
+    return ReachabilityEvent(id=42, router="example-gw.example.org", state=EventState.OPEN)
+
+
+@pytest.fixture
+def changed_fake_event(fake_event) -> ReachabilityEvent:
+    copy = fake_event.model_copy(deep=True)
+    copy.add_history("this fake event has been changed")
+    return copy
```

Comment:
Do we actually need to make a fixture out of it if it is only used once?

Reply:
No, but I like having it around, as the explicit naming and separation of concerns appeal to me.